sklearn/utils/estimator_checks.py

```diff
@@ -539,6 +548,25 @@ def check_classifiers_train(name, Classifier):
             assert_raises(ValueError, classifier.predict_proba, X.T)
 
 
+def check_estimators_unfitted(name, Estimator):
+    """Check if NotFittedError is raised when calling predict and related
+    functions"""
+
+    # Common test for Regressors as well as Classifiers
+    X, y = _boston_subset()
+
+    with warnings.catch_warnings(record=True):
+        est = Estimator()
+
+    assert_raises(NotFittedError, est.predict, X)
+    
+    if hasattr(est, 'decision_function'):
+        assert_raises(NotFittedError, est.decision_function, X)
+
+    if hasattr(est, 'predict_proba'):
+        assert_raises(NotFittedError, est.predict_proba, X)
```

Comment:
> I don't think `predict_log_proba` is so necessary here. 
Do I remove that test?

> Ideally more obscure methods like kneighbors on nearest neighbors estimators should also follow this API
Would it be more appropriate, if those tests were included under respective tests for those modules?